adaptor-rm-webapp/src/main/java/com/sample/rm/RMToolManager.java

```diff
@@ -217,26 +304,78 @@ public static Requirement getRequirement(HttpServletRequest httpServletRequest,
+        } catch (NoSuchElementException e) {
+            log.error("Resource: '" + uri + "' not found");
+            throw new WebApplicationException("Failed to get resource: '" + uri + "'", e, Status.NOT_FOUND);
+        } catch (StoreAccessException | ModelUnmarshallingException  e) {
+            log.error("Failed to get resource: '" + uri + "'", e);
+            throw new WebApplicationException("Failed to get resource: '" + uri + "'", e, Status.INTERNAL_SERVER_ERROR);
+        } finally {
+            storePool.releaseStore(store);
+        }
+        // Start of user code getRequirement_storeFinalize
         // End of user code
+        
+        // Start of user code getRequirement
+       // End of user code
         return aResource;
     }
 
     public static Boolean deleteRequirement(HttpServletRequest httpServletRequest, final String requirementId)
     {
         Boolean deleted = false;
+        // Start of user code deleteRequirement_storeInit
+        // End of user code
+        Store store = storePool.getStore();
+        URI uri = RMToolResourcesFactory.constructURIForRequirement(requirementId);
+        if (!store.resourceExists(storePool.getDefaultNamedGraphUri(), uri)) {
+            log.error("Cannot delete a resource that does not already exists: '" + uri + "'");
+            throw new WebApplicationException("Cannot delete a resource that does not already exists: '" + uri + "'", Status.NOT_FOUND);
+        }
+        store.deleteResources(storePool.getDefaultNamedGraphUri(), uri);
+        storePool.releaseStore(store);
+        deleted = true;
+        // Start of user code deleteRequirement_storeFinalize
+        // End of user code
+        
         // Start of user code deleteRequirement
-        // TODO Implement code to delete a resource
-        // If you encounter problems, consider throwing the runtime exception WebApplicationException(message, cause, final httpStatus)
+        trsEventHandler.onDeleted(uri);
         // End of user code
         return deleted;
     }
 
     public static Requirement updateRequirement(HttpServletRequest httpServletRequest, final Requirement aResource, final String requirementId) {
         Requirement updatedResource = null;
+        // Start of user code updateRequirement_storeInit
+        // End of user code
+        Store store = storePool.getStore();
+        URI uri = RMToolResourcesFactory.constructURIForRequirement(requirementId);
+        if (!store.resourceExists(storePool.getDefaultNamedGraphUri(), uri)) {
+            log.error("Cannot update a resource that does not already exists: '" + uri + "'");
+            throw new WebApplicationException("Cannot update a resource that does not already exists: '" + uri + "'", Status.NOT_FOUND);
+        }
+        aResource.setAbout(uri);
+        try {
+            store.deleteResources(storePool.getDefaultNamedGraphUri(), uri);
```

Comment:
https://github.com/eclipse/lyo.store/blob/master/src/store-core/src/main/java/org/eclipse/lyo/store/internals/SparqlStoreImpl.java#L340 is better, we might optimize it later to delete and insert in a single SPARQL query.

Reply:
Yes. we need to both optimize, but also make it thread-safe.

Btw, It's confusing with 
* updateResources vs putResources
* appendResource vs InsertResurces.
But in the end, we are doing the same thing.